packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts

```diff
@@ -0,0 +1,275 @@
+import {Schema} from '@coveo/bueno';
+import {AsyncThunkAction} from '@reduxjs/toolkit';
+import {
+  ClickAction,
+  CustomAction,
+} from '../../../features/analytics/analytics-utils';
+import {configurationReducer as configuration} from '../../../features/configuration/configuration-slice';
+import {
+  foldingOptionsSchemaDefinition,
+  registerFolding,
+} from '../../../features/folding/folding-actions';
+import {foldingReducer as folding} from '../../../features/folding/folding-slice';
+import {
+  CollectionId,
+  FoldedCollection,
+  FoldedResult,
+} from '../../../features/folding/folding-state';
+import {queryReducer as query} from '../../../features/query/query-slice';
+import {searchReducer as search} from '../../../features/search/search-slice';
+import {CoreEngine, Result} from '../../../recommendation.index';
+import {
+  ConfigurationSection,
+  FoldingSection,
+  QuerySection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {validateOptions} from '../../../utils/validate-payload';
+import {Controller} from '../../controller/headless-controller';
+import {buildResultList} from '../../result-list/headless-result-list';
+import {ResultListOptions} from '../result-list/headless-core-result-list';
+import {SearchStatusState} from '../status/headless-core-status';
+
+export type {FoldedCollection, FoldedResult};
+
+const optionsSchema = new Schema<Required<FoldingOptions>>(
+  foldingOptionsSchemaDefinition
+);
+
+export interface FoldingOptions {
+  /**
+   * The name of the field on which to do the folding. The folded result list component will use the values of this field to resolve the collections of result items.
+   *
+   * @defaultValue `foldingcollection`
+   */
+  collectionField?: string;
+  /**
+   * The name of the field that determines whether a certain result is a top result containing other child results within a collection.
+   *
+   * @defaultValue `foldingparent`
+   */
+  parentField?: string;
+  /**
+   * The name of the field that uniquely identifies a result within a collection.
+   *
+   * @defaultValue `foldingchild`
+   */
+  childField?: string;
+  /**
+   * The number of child results to fold under the root collection element, before expansion.
+   *
+   * @defaultValue `2`
+   */
+  numberOfFoldedResults?: number;
+}
+
+export interface FoldedResultListOptions extends ResultListOptions {
+  folding?: FoldingOptions;
+}
+
+export interface FoldedResultListProps {
```

Comment:
You  need to name this `CoreFoldedResultListProps` because that's not the interface that we need to pass to the search/insight folding result controller.
The interface we need to pass to the search/insight folding result controller doesn’t need to contain `loadCollectionActionCreator`, it will just be: 
```
export interface FoldedResultListProps {
  /**
   * The options for the `FoldedResultList` controller.
   * */
  options?: FoldedResultListOptions;
}
```

Reply:
Yea okay if I understand what you told me correctly Simon and I don't remember why we need that in the props there, but @mmitiche is correct.

Name this one `CoreFoldedResultListProps`